core/src/Cabana_LinkedCellList.hpp

```diff
@@ -204,34 +208,45 @@ class LinkedCellList
     */
     BinningData<DeviceType> binningData() const { return _bin_data; }
 
-  public:
-    // This function should be private but we need to expose it as public to
-    // launch CUDA kernels with class data.
+    /*!
+      \brief Build the linked cell list with a subset of particles.
+
+      \tparam SliceType Slice type for positions.
+
+      \param positions Slice of positions.
+
+      \param begin The beginning index of the slice range to sort.
+
+      \param end The end index of the slice range to sort.
+    */
     template <class SliceType>
     void build( SliceType positions, const std::size_t begin,
                 const std::size_t end )
     {
-        // Allocate the binning data. Note that the permutation vector spans
+        // Resize the binning data. Note that the permutation vector spans
         // only the length of begin-end;
         std::size_t ncell = totalBins();
-        Kokkos::View<int*, memory_space> counts(
-            Kokkos::view_alloc( Kokkos::WithoutInitializing, "counts" ),
-            ncell );
-        OffsetView offsets(
-            Kokkos::view_alloc( Kokkos::WithoutInitializing, "offsets" ),
-            ncell );
-        OffsetView permute(
-            Kokkos::view_alloc( Kokkos::WithoutInitializing, "permute" ),
-            end - begin );
+        if ( _counts.extent( 0 ) != ncell )
+        {
+            Kokkos::resize( _counts, ncell );
```

Comment:
I think this is fine for now because `Kokkos::resize` doesn't do anything if the grid size is the same. In the future if we have dynamic grid sizes we could keep a separate `size` and `capacity` here and only resize when we need more allocation.